java_src/foam/dao/index/Index.java

```diff
@@ -18,6 +19,7 @@
   public Object put(Object state, FObject value);
   public Object remove(Object state, FObject value);
   public Object removeAll();
+  public Object get(Object state, Object key);
```

Comment:
Why is this needed? Maybe there should be a sub-interface called PrimaryIndex which has this extra method.